core/src/main/java/io/kestra/core/services/AbstractFlowTriggerService.java

```diff
@@ -0,0 +1,158 @@
+package io.kestra.core.services;
+
+import io.kestra.core.models.conditions.types.MultipleCondition;
+import io.kestra.core.models.executions.Execution;
+import io.kestra.core.models.flows.Flow;
+import io.kestra.core.models.triggers.AbstractTrigger;
+import io.kestra.core.models.triggers.multipleflows.MultipleConditionStorageInterface;
+import io.kestra.core.models.triggers.multipleflows.MultipleConditionWindow;
+import io.kestra.core.runners.RunContextFactory;
+import jakarta.inject.Inject;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.ToString;
+
+import java.util.*;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public abstract class AbstractFlowTriggerService<S extends MultipleConditionStorageInterface> {
+    @Inject
+    protected Optional<S> multipleConditionStorage;
+
+    @Inject
+    private ConditionService conditionService;
+
+    @Inject
+    private RunContextFactory runContextFactory;
+
+    @Inject
+    private FlowService flowService;
+
+    public AbstractFlowTriggerService(Optional<S> multipleConditionStorage, ConditionService conditionService, RunContextFactory runContextFactory, FlowService flowService) {
+        this.multipleConditionStorage = multipleConditionStorage;
+        this.conditionService = conditionService;
+        this.runContextFactory = runContextFactory;
+        this.flowService = flowService;
+    }
```

Comment:
Please choose between field injection or constructor injection but don't use both.

Reply:
Then I'll have to stick with constructor as Kafka provides its multipleConditionStorage based on a State Store which is a runtime thing :'(